core/types/receipt.go

```diff
@@ -179,12 +180,12 @@ func (r *ReceiptForStorage) DecodeRLP(s *rlp.Stream) error {
 		return err
 	}
 
-	// Try decoding the receipt without Status first
-	if err := decodeStoredReceiptRLP(r, raw); err == nil {
+	// Try decoding the receipt with Status first
+	if err := decodeStoredReceiptRLPv2(r, raw); err == nil {
```

Comment:
`...v2` or similar suffixes makes a lot of confusion.
It's much easier to have meaningfull domain name, especially when it's not bound to any actuall versioning.

Reply:
Yeah it was just already a long variable name, any suggestions then? Because it isn't tied to any consensus or protocol change either.

Reply:
```suggestion
	if err := decodeStoredReceiptRLPWithStatus(r, raw); err == nil {
```

length doesn't matter, readability does

Reply:
ok I'll do that now then if no other changes yet